CI failure:
```
D:\buildtrees\ada-url\src\v2.4.1-cb166651e1.clean\tools\cli\adaparse.cpp(223): error C2039: 'output_file': is not a member of 'fmt'
D:\buildtrees\ada-url\src\v2.4.1-cb166651e1.clean\dependencies\.cache\fmt\include\fmt/os.h(73): note: see declaration of 'fmt'
D:\buildtrees\ada-url\src\v2.4.1-cb166651e1.clean\tools\cli\adaparse.cpp(223): error C3861: 'output_file': identifier not found
```
Error log: https://dev.azure.com/vcpkg/public/_build/results?buildId=89565&view=artifacts&pathAsName=false&type=publishedArtifacts

--- a/ports/ada-url/portfile.cmake
+++ b/ports/ada-url/portfile.cmake
@@ -2,8 +2,14 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO ada-url/ada
     REF "v${VERSION}"
-    SHA512 a27fc3f8713e73baacbec4f36688bb60c71cdde0ad6142dfe314d09385757833e6166995a955856620183f6fe9bf04d7e4d62e3f3de11c6af86d2d988b719729
+    SHA512 689dcdc3e89f2ae51e039f6d354efea3231d3ff025c58f572596d87cebd7d5f2bf5ca8c530df8514369a14831dfdb1c65bc1d93a0ece579031eb3f0dd47548c0
     HEAD_REF main
+)
+
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        tools ADA_TOOLS
 )
 
 vcpkg_cmake_configure(
@@ -12,6 +18,7 @@ vcpkg_cmake_configure(
     OPTIONS
         -DADA_BENCHMARKS=OFF
         -DBUILD_TESTING=OFF
+        ${FEATURE_OPTIONS}
 )
 
 vcpkg_cmake_install()
@@ -20,7 +27,9 @@ vcpkg_copy_pdbs()
 
 vcpkg_cmake_config_fixup(PACKAGE_NAME ada CONFIG_PATH "lib/cmake/ada")
 
-vcpkg_copy_tools(TOOL_NAMES adaparse AUTO_CLEAN)
+if("tools" IN_LIST FEATURES)
+    vcpkg_copy_tools(TOOL_NAMES adaparse AUTO_CLEAN)
+endif()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 

--- a/ports/ada-url/vcpkg.json
+++ b/ports/ada-url/vcpkg.json
@@ -16,7 +16,8 @@
   ],
   "features": {
     "tools": {
-      "description": "Build CLI tools (adaparse)"
+      "description": "Build CLI tools (adaparse)",
+      "supports": "!uwp"
     }
   }
 }

--- a/ports/ada-url/vcpkg.json
+++ b/ports/ada-url/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "ada-url",
-  "version": "2.4.1",
+  "version": "2.4.2",
   "description": "WHATWG-compliant and fast URL parser written in modern C++",
   "homepage": "https://ada-url.com/",
   "license": "MIT",
@@ -13,5 +13,10 @@
       "name": "vcpkg-cmake-config",
       "host": true
     }
-  ]
+  ],
+  "features": {
+    "tools": {
+      "description": "Build CLI tools (adaparse)"
+    }
+  }
 }

--- a/versions/a-/ada-url.json
+++ b/versions/a-/ada-url.json
@@ -1,8 +1,8 @@
 {
   "versions": [
     {
-      "git-tree": "9c379cd4c6b72ab63c4fe29a4b69ef3cd2600d89",
-      "version": "2.4.1",
+      "git-tree": "fb3ba1bf37356f538db56db9af440338cc8de761",
+      "version": "2.4.2",
       "port-version": 0
     }
   ]

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -37,7 +37,7 @@
       "port-version": 12
     },
     "ada-url": {
-      "baseline": "2.4.1",
+      "baseline": "2.4.2",
       "port-version": 0
     },
     "ade": {

